plugins/SolidView/SolidView.py

```diff
@@ -27,12 +49,31 @@ def __init__(self):
         self._non_printing_shader = None
         self._support_mesh_shader = None
 
+        self._xray_shader = None
+        self._xray_pass = None
+        self._xray_composite_shader = None
+        self._composite_pass = None
+        self._xray_error_image = None
+
         self._extruders_model = None
         self._theme = None
         self._support_angle = 90
 
         self._global_stack = None
 
+        self._old_composite_shader = None
+        self._old_layer_bindings = None
+
+        self._next_xray_checking_time = time.time()
+        self._xray_checking_update_time = 1.0 # seconds
+        self._xray_warning_cooldown = 60 * 10 # reshow Model error message every 10 minutes
+        self._xray_warning_message = Message(catalog.i18nc("@info:status", "Your model is not manifold. The highlighted areas indicate either missing or extraneous surfaces.")
+                                             , lifetime = 60 * 5 # leave message for 5 minutes
```

Comment:
The only reason I set a finite timer to it was so that it reshows itself after five minutes when you move the camera. That reshowing can draw more attention to it.

I think the user really needs to se this message. He will continually be reminded of it until he expressly sayd "Do not show again".